src/System.Text.Json/src/System/Text/Json/Serialization/JsonSnakeCaseNamingPolicy.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSnakeCaseNamingPolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
```

Comment:
Any particular reason to use a different implementation than what was done in Newtonsoft.Json (which this PR had: https://github.com/dotnet/corefx/pull/40003)?

Reply:
Npgsql used the same code as Newtonsoft.Json, but it had a bug which I fixed. See npgsql/npgsql#2152.